op-challenger/runner/runner.go

```diff
@@ -115,20 +108,58 @@ func (r *Runner) loop(ctx context.Context, traceType types.TraceType, client *so
 	}
 }
 
-func (r *Runner) runOnce(ctx context.Context, traceType types.TraceType, client *sources.RollupClient, caller *batching.MultiCaller) error {
+func (r *Runner) runAndRecordOnce(ctx context.Context, traceType types.TraceType, client *sources.RollupClient, caller *batching.MultiCaller) {
+	recordError := func(err error, traceType string, m Metricer, log log.Logger) {
+		if errors.Is(err, ErrUnexpectedStatusCode) {
+			log.Error("Incorrect status code", "type", traceType, "err", err)
+			m.RecordInvalid(traceType)
+		} else if err != nil {
+			log.Error("Failed to run", "type", traceType, "err", err)
+			m.RecordFailure(traceType)
+		} else {
+			log.Info("Successfully verified output root", "type", traceType)
+			m.RecordSuccess(traceType)
+		}
+	}
+
 	prestateHash, err := r.getPrestateHash(ctx, traceType, caller)
 	if err != nil {
-		return err
+		recordError(err, traceType.String(), r.m, r.log)
+		return
 	}
 
 	localInputs, err := r.createGameInputs(ctx, client)
 	if err != nil {
-		return err
+		recordError(err, traceType.String(), r.m, r.log)
+		return
 	}
 	dir, err := r.prepDatadir(traceType)
 	if err != nil {
-		return err
+		recordError(err, traceType.String(), r.m, r.log)
+		return
 	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		err = r.runOnce(ctx, traceType, prestateHash, localInputs, dir)
+		recordError(err, traceType.String(), r.m, r.log)
+		wg.Done()
+	}()
+
+	if r.addMTCannonPrestate != (common.Hash{}) {
+		wg.Add(1)
+		go func() {
+			// reuse the trace directory
```

Comment:
Now that these are running in parallel, do you still want to reuse this directory?

Reply:
No, will fix. I originally intended to speed up MT-Cannon trace generation by reusing the singlethreaded preimages. But this breaks if multiple hosts use the same pebble kv store. While pebble isn't the default kvstore, I'd rather future-proof the runner in case that changes.